src/scp/NominationProtocol.cpp

```diff
@@ -664,4 +817,19 @@ NominationProtocol::getLatestMessage(NodeID const& id) const
     }
     return nullptr;
 }
+
+static const int NOMINATION_MAX_TIMEOUT_SECONDS = (30 * 60); //same value as MAX_TIMEOUT_SECONDS defined within SCPDriver.cpp
+std::chrono::milliseconds
+NominationProtocol::computeNominationTimeout(uint32 roundNumber)
+{
+    if (roundNumber > NOMINATION_MAX_TIMEOUT_SECONDS)
```

Comment:
This doesn't seem like it would work correctly since each round only adds 25% of the time that it did before.

Reply:
You're right.  The term for upper bound check should be leaderNominationTimeoutInMilisec (incremented value for leader selection timeout).

Reply:
Shouldn't it be `NOMINATION_MAX_TIMEOUT_SECONDS * 1000 / BASE_LEADER_NOMINATION_TIMEOUT_MS`?